pkg/pmem-csi-driver/pmem-csi-driver.go

```diff
@@ -317,6 +317,13 @@ func (pmemd *pmemDriver) Run() error {
 		// Here we want to shut down cleanly, i.e. let running
 		// gRPC calls complete.
 		klog.V(3).Infof("Caught signal %s, terminating.", sig)
+		if pmemd.cfg.Mode == Node {
+			klog.V(3).Info("Unregistering node...")
+			if err := pmemd.unregisterNodeController(); err != nil {
```

Comment:
Can it happen that this attempt to unregister slows down the shutdown when the controller happens to exit first?

Reply:
Yes, it does...

https://cloudnative-k8sci.southcentralus.cloudapp.azure.com/blue/rest/organizations/jenkins/pipelines/pmem-csi/branches/version-skew-2/runs/1/nodes/97/steps/201/log/?start=0

```
[2020-09-08T22:10:21.367Z] pmem-csi-controller-0/pmem-driver@pmem..ster: I0908 22:10:21.209248       1 pmem-csi-driver.go:296] Caught signal terminated, terminating.
[2020-09-08T22:10:21.367Z] pmem-csi-controller-0/pmem-driver@pmem..ster: I0908 22:10:21.215828       1 server.go:55] Server on 'unix:///csi/csi-controller.sock' stopped
[2020-09-08T22:10:21.367Z] pmem-csi-node-zrcfj/pmem-driver@pmem..ker2: I0908 22:10:21.218459       1 pmem-csi-driver.go:468] Connection state: CONNECTING
[2020-09-08T22:10:21.367Z] pmem-csi-node-zrcfj/pmem-driver@pmem..ker2: I0908 22:10:21.223549       1 pmem-csi-driver.go:468] Connection state: TRANSIENT_FAILURE
[2020-09-08T22:10:21.367Z] pmem-csi-node-mvvzx/pmem-driver@pmem..ker1: I0908 22:10:21.225583       1 pmem-csi-driver.go:468] Connection state: CONNECTING
[2020-09-08T22:10:21.367Z] pmem-csi-controller-0/pmem-driver@pmem..ster: I0908 22:10:21.231423       1 server.go:55] Server on 'tcp://0.0.0.0:10000' stopped
[2020-09-08T22:10:21.367Z] pmem-csi-controller-0/pmem-driver@pmem..ster: PASS
...
[2020-09-08T22:10:37.050Z] Sep  8 22:10:37.018: INFO: waiting for deletion of pmem-csi-node-jp5kd (v1.Pod, 855cb2bf-ebfc-4fb6-abce-2a9991f05191)
[2020-09-08T22:10:37.050Z] Sep  8 22:10:37.018: INFO: waiting for deletion of pmem-csi-node-mvvzx (v1.Pod, 7254a27f-eccd-4079-a1cf-9c5707f1c98d)
[2020-09-08T22:10:37.050Z] Sep  8 22:10:37.018: INFO: waiting for deletion of pmem-csi-node-zrcfj (v1.Pod, 4462c334-fae4-41a0-99ab-54634440cef2)
...
[2020-09-08T22:10:41.715Z] pmem-csi-node-jp5kd/pmem-driver@pmem..ker3: E0908 22:10:41.395483       1 pmem-csi-driver.go:300] Failed to unregister node: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 10.105.63.129:10000: i/o timeout"
[2020-09-08T22:10:41.715Z] pmem-csi-node-jp5kd/pmem-driver@pmem..ker3: failed to run driver: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 10.105.63.129:10000: i/o timeout"
[2020-09-08T22:10:41.715Z] pmem-csi-node-jp5kd/pmem-driver@pmem..ker3: PASS
```